src/uct/cuda/cuda_ipc/cuda_ipc_ep.c

```diff
@@ -31,16 +35,151 @@ UCS_CLASS_DEFINE_NEW_FUNC(uct_cuda_ipc_ep_t, uct_ep_t, uct_iface_t*,
                           const uct_device_addr_t *, const uct_iface_addr_t *);
 UCS_CLASS_DEFINE_DELETE_FUNC(uct_cuda_ipc_ep_t, uct_ep_t);
 
-ucs_status_t uct_cuda_ipc_ep_get_zcopy(uct_ep_h tl_ep, const uct_iov_t *iov, size_t iovcnt,
-                                       uint64_t remote_addr, uct_rkey_t rkey,
-                                       uct_completion_t *comp)
+#define uct_cuda_ipc_trace_data(_remote_addr, _rkey, _fmt, ...)     \
+    ucs_trace_data(_fmt " to %"PRIx64"(%+ld)", ## __VA_ARGS__, (_remote_addr), \
+                   (_rkey))
+
+#define UCT_CUDA_IPC_ZERO_LENGTH_POST(len)                      \
+    do {                                                        \
+        if (0 == len) {                                         \
+            ucs_trace_data("Zero length request: skip it");     \
+            return UCS_OK;                                      \
+        }                                                       \
+    } while(0);
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+uct_cuda_ipc_get_mapped_addr(uct_cuda_ipc_key_t *key, int cu_device,
+                             uint64_t remote_addr, void *mapped_rem_addr,
+                             const uct_iov_t *iov, uct_cuda_ipc_ep_t *ep,
+                             uct_cuda_ipc_iface_t *iface)
 {
+    void *mapped_rem_base_addr = NULL;
```

Comment:
this also output from cuIpcOpenMemHandle(). I think no need to assign

Reply:
fixed